components/activity_and_insights/insights/top_dms_and_new_members/new_members_item/new_members_item.tsx

```diff
@@ -30,6 +32,21 @@ type Props = {
 
 const NewMembersItem = ({newMember, team}: Props) => {
     const teammateNameDisplaySetting = useSelector(getTeammateNameDisplaySetting);
+    const user = useSelector((state: GlobalState) => getUser(state, newMember.id)) as UserProfile;
+    const [member, setMember] = useState<UserProfile>(newMember as UserProfile);
+
+    useEffect(() => {
+        if (user) {
+            setMember({
+                ...member,
+                username: user.username,
+                first_name: user.first_name,
+                last_name: user.last_name,
+                nickname: user.nickname,
+                last_picture_update: user.last_picture_update,
+            });
+        }
+    }, [user]);
```

Comment:
0/5 but setting similar state variable based off on similar type props leads to bugs which are hard to track (no i do not have any proof of that 🤭 , faced it before).
We dont event need use effect here if we just make a new variable called member

```js
const member = ({...newMember, username: user.username ......})
```
and let rendering update the variable automatically.

Another reason is because use of `setMember` only seem to be used in one useEffect. if you plan to use else where to please ignore my suggestion.

Reply:
Yes, you're probably right. I didn't even like using the state just for that, so I'll give your suggestion a try – thanks!